opr/assetList.go

```diff
@@ -17,9 +23,80 @@ func (o OraclePriceRecordAssetList) Contains(list []string) bool {
 			return false
 		}
 	}
+
 	return true
 }
 
+// Exchange tells us how much we need to spend given the amount we want is fixed.
+//	?? FROM -> X TO
+//
+//   X TO         to_usd               1
+//  ------    *  --------- = USD * -------- = FROM
+//     1            1               from_usd
+//
+func (o OraclePriceRecordAssetList) ExchangeTo(from string, to string, want int64) (int64, error) {
+	fromRate, toRate, err := o.ExchangeRates(from, to)
+	if err != nil {
+		return 0, err
+	}
+
+	// Truncate vs round, so txs will drop the partials, as rounding up would create tokens.
+	// Office space code goes here
+	return common.TruncateFloat(float64(want) * toRate / fromRate), err
+}
+
+// Exchange tells us how much we need to spend given the amount we have is fixed.
+//  X FROM -> ?? TO
+//
+//  X FROM       from_usd             1
+//  ------    *  --------- = USD * -------- = TO
+//     1            1               to_usd
+//
+func (o OraclePriceRecordAssetList) ExchangeFrom(from string, have int64, to string) (int64, error) {
+	fromRate, toRate, err := o.ExchangeRates(from, to)
+	if err != nil {
+		return 0, err
+	}
+
+	// Truncate vs round, so txs will drop the partials, as rounding up would create tokens.
+	// Office space code goes here
+	return common.TruncateFloat(float64(have) * fromRate / toRate), err
+}
+
+// ExchangeRate finds the exchange rates for FROM and TO in usd as the base pair.
+func (o OraclePriceRecordAssetList) ExchangeRates(from, to string) (fromRate float64, toRate float64, err error) {
+	var ok bool
+	// First we need to ensure we have the pricing for each side of the exchange
+	fromRate, ok = o[from]
+	if !ok {
+		return 0, 0, fmt.Errorf("did not find a rate for %s", from)
+	}
+
+	toRate, ok = o[to]
+	if !ok {
+		return 0, 0, fmt.Errorf("did not find a rate for %s", to)
+	}
+
+	if toRate == 0 || fromRate == 0 {
+		return 0, 0, fmt.Errorf("one of the rates found is 0")
+	}
+
+	// fromRate / toRate
+	return fromRate, toRate, nil
+}
+
+// ExchangeRate finds the exchange rate going from `FROM` to `TO`.
+//	To do the exchange rate, USD is the base pair and used as the intermediary.
+//	So to go from FCT -> BTC, the math goes:
+//		FCT -> USD -> BTC
+func (o OraclePriceRecordAssetList) ExchangeRate(from, to string) (float64, error) {
```

Comment:
this function is not compatible with the above ExchangeTo/From. the above functions multiply first, then divide, whereas this function divides first, meaning you get different results

assuming we have:
1 FTC = 8 USD
1 BTC = 123456.12345678 USD

and we want to convert 6,838,893.09091930 FTC to BTC. 

```go
o := make(opr.OraclePriceRecordAssetList)
o["USD"] = 1
o["FCT"] = 8
o["BTC"] = 123456.12345678

fmt.Println(o.ExchangeFrom("FCT", 683889309091930, "BTC"))
rate, _ := o.ExchangeRate("FCT", "BTC")
fmt.Println(common.TruncateFloat(float64(683889309091930) * rate))
```
the result is:
```
44316266537
44316266536
```
meaning a 1 sat difference. we really need to use a dedicated library for this kind of math